homeassistant/components/dwd_weather_warnings/sensor.py

```diff
@@ -127,15 +128,14 @@ class DwdWeatherWarningsSensor(SensorEntity):
     def __init__(
         self,
         api,
-        name,
-        unique_id,
+        entry: ConfigEntry,
         description: SensorEntityDescription,
     ) -> None:
         """Initialize a DWD-Weather-Warnings sensor."""
-        self._api = api
         self.entity_description = description
-        self._attr_name = f"{name} {description.name}"
-        self._attr_unique_id = f"{unique_id}-{description.key}"
+        self._attr_name = f"{DEFAULT_NAME} {entry.title} {description.name}"
```

Comment:
I think you can introduce `DeviceInfo` and also use `has_entity_name` in this PR.

Reply:
And removing `self._attr_name` right? Currently testing this, had to set the `entity_id` manually because it didn't include the `dwd_weather_warnings` (probably because I used the config entry title for the device info name). The friendly name doesn't have the `DWD Weather Warnings` anymore but that should be fine

Reply:
I suppose the device namn can be `f"{DEFAULT_NAME} {entry.title}` and `attr_name` could be `description.name` which together then would construct the full name (and entity id) if you set `has_entity_name`

Reply:
Let's keep entity names out of this PR and do that in a separate PR to review.